src/Voron/Data/BTrees/TreePageSplitter.cs

```diff
@@ -73,7 +74,7 @@ private FreeSpaceHandlingDisabler DisableFreeSpaceUsageIfSplittingRootTree()
 
                 if (_page.IsCompressed)
                 {
-                    _pageDecompressed = _tree.DecompressPage(_page);
+                    _pageDecompressed = _tree.DecompressPage(_page, WriteDecompressionUsage, skipCache: false);
```

Comment:
The fix is here - we need to pass `DecompressionUsage.Write` since we're about to split that page